backend/de.metas.monitoring/src/main/java/de/metas/monitoring/adapter/MicrometerPerformanceMonitoringService.java

```diff
@@ -124,22 +143,38 @@ private void appendLabelsToTags(@NonNull final ArrayList<Tag> tags, @NonNull fin
 		}
 	}
 	
-	private void addTagIfNotNull(@Nullable final String name, @Nullable final String value, @NonNull ArrayList<Tag> tags)
+	private static void addTagIfNotNull(@Nullable final String name, @Nullable final String value, @NonNull ArrayList<Tag> tags)
 	{
 		if (!EmptyUtil.isBlank(name) && !EmptyUtil.isBlank(value))
 		{
 			tags.add(Tag.of(name, value));
 		}
 	}
 
-	private void incrementDepth(PerformanceMonitoringData perfMonData)
+	private static void addInitiatorTags(
+			@NonNull PerformanceMonitoringData perfMonData,
+			@NonNull Metadata metadata,
+			@NonNull ArrayList<Tag> tags)
 	{
-		perfMonData.setDepth(perfMonData.getDepth() + 1);
+		addTagIfNotNull("depth", String.valueOf(perfMonData.getDepth()), tags);
+		addTagIfNotNull("initiator", perfMonData.getInitiator(), tags);
+		addTagIfNotNull("window", perfMonData.getInitiatorWindow(), tags);
+		addTagIfNotNull("callerName", metadata.getClassName() + " - " + metadata.getFunctionName(), tags);
+		addTagIfNotNull("calledBy", perfMonData.getCalledBy().get(perfMonData.getDepth()), tags);
 	}
 
-	private void decrementDepth(PerformanceMonitoringData perfMonData)
+	private Type getTypeForMeterName(Metadata metadata)
```

Comment:
in general the type is used for the meter-name (meter = set of metrics),
but all functions called by an RestController-function are grouped in Type.REST_API_PROCESSING for dashboard use (so for example type.db gets replaced).
So this function replaces the type or keeps the restController type

Reply:
hmmm.... this is smelly. 

why not group everything under the first caller? I mean, the first caller can tell if is a REST API or not.

For me this method does not make sense.
1. the name is wrong... has nothing to do with what it does
2. the logic is "all about REST"  (i.e. if the type is something related to REST then OK keep it if not, i don't care, i consider is about REST).... and that's wrong.

You shall know if is a rest or NOT if the initiator is REST or not. 

am I missing something?

cc @metas-ts

Reply:
>why not group everything under the first caller?
this would make the use of `Prometheus Query Language` in the Grafana Dashboard much more complicated and they would take longer. RestController metrics and underlaying functions metrics are handled separately.

>1. the name is wrong... has nothing to do with what it does
- the meter name is set like METER_PREFIX + type.getCode(), which results in mf_rest_controller...
- without this function every Type would get it's own name (like mf_PO, mf_DB, mf_modelInterceptor...)
- so this function gives me the type that should be used for the name
  - RestController functions keep their name
  - underlying functions get mf_rest_api_processing as name
  - functions not called by a rest controller keep their name mf_db, mf_po (else clause: `!perfMonData.isInitiatorLabelActive()`)